### PR TITLE
Get 100% test coverage for Giphy bot.

### PR DESCRIPTION
**Pull request by roberthoenig**
This comes along with a few improvements to the testing system (and bot improvements).

### CODE REVIEW

**Comment by roberthoenig** [comment]
This fails because of Python 2 incompatibility. I'm gonna update Travis tests so `zulip_bots` is excluded from the Python 2 tests.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 language: python
 python:
-  - "2.7"
   - "3.4"
   - "3.5"
   - "3.6"
 env: TEST_SUITE=test-main
 matrix:
   include:
+  - python: "2.7"
```
Comment (showell) [inline .travis.yml]: Do we actually need this?  If so, I would clarify the commit message.

Comment (timabbott) [reply .travis.yml]: Looks correct to me.

Comment (timabbott) [reply .travis.yml]: (But I think the commit message could benefit here, yeah)

File: zulip_bots/zulip_bots/test_lib.py [hunk 2]
```diff
@@ -136,12 +136,9 @@ def mock_http_conversation(self, test_name):
             http_response = http_data.get('response')
             http_headers = http_data.get('response-headers')
             with patch('requests.get') as mock_get:
-                mock_result = mock.MagicMock()
-                mock_result.json.return_value = http_response
-                if 'text' in http_response:
-                    mock_result.text = http_response.get('text', None)
+                mock_result = requests.Response()
```
Comment (showell) [inline zulip_bots/zulip_bots/test_lib.py]: I would explain with a comment how it's safe to use a "real" `Response` object in a `mock_result` context.

**Comment by timabbott** [comment]
@eeshangarg FYI, if we're dropping Python 2 support for zulip_bots and zulip_botserver, there will be corresponding packaging changes we want to make (I'd just removing the packaging tags, and not do a bunch of migrations to python-3-only stuff, for now).

**Comment by showell** [comment]
Ok. I merged this.  I'm a little unsure on a couple things, but I don't want this to get stale when Robert's on vacation and when I get pulled away to other stuff.  If my concerns are valid, we can always fix forward.

Thanks @roberthoenig!

**Comment by timabbott** [comment]
The main concern is that we need to update the release tools to not try releasing a Python 2.7 copy.

**Pull request merged**